apm-sniffer/apm-sdk-plugin/graphql-plugin/graphql-9.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/graphql/v9/GraphqlInterceptor.java

```diff
@@ -0,0 +1,78 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.graphql.v9;
+
+import graphql.execution.ExecutionPath;
+import graphql.execution.ExecutionStrategyParameters;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+public class GraphqlInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+        ExecutionStrategyParameters parameters = (ExecutionStrategyParameters) allArguments[1];
+        if (parameters == null || parameters.getParent().getPath() != ExecutionPath.rootPath()) {
+            return;
+        }
+        AbstractSpan span = ContextManager.createLocalSpan(parameters.getField().get(0).getName());
+        Tags.LOGIC_ENDPOINT.set(span, buildLogicEndpointSpan());
+        span.setComponent(ComponentsDefine.GRAPHQL);
+    }
+
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, Object ret) throws Throwable {
+        ExecutionStrategyParameters parameters = (ExecutionStrategyParameters) allArguments[1];
+        if (parameters == null || parameters.getParent().getPath() != ExecutionPath.rootPath()) {
+            return ret;
+        }
+        ContextManager.stopSpan();
+        return ret;
+    }
+
+    @Override
+    public void handleMethodException(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, Throwable t) {
+        ExecutionStrategyParameters parameters = (ExecutionStrategyParameters) allArguments[1];
+        if (parameters == null || parameters.getParent().getPath() != ExecutionPath.rootPath()) {
+            return;
+        }
+        dealException(t);
+    }
+
+    private void dealException(Throwable throwable) {
+        AbstractSpan span = ContextManager.activeSpan();
+        span.errorOccurred();
+        span.log(throwable);
+    }
+
+    private String buildLogicEndpointSpan() {
+        Map<String, Object> logicEndpointSpan = new HashMap<>();
+        logicEndpointSpan.put("logic-span", true);
+        return logicEndpointSpan.toString();
```

Comment:
The value of `x-le` should be JSON format, this is obviously not the case by simply calling `toString` on a `Map`